pymodbus/server/async.py

```diff
@@ -274,9 +274,23 @@ def StartSerialServer(context, identity=None,
     SerialPort(protocol, port, reactor, baudrate)
     reactor.run()
 
+
+def _stop_server(is_mainthread=True):
+    from twisted.internet import reactor
+    if _is_main_thread():
+        reactor.stop()
+    else:
+        reactor.callFromThread(reactor.stop)
+    _logger.debug("Stopping main thread")
```

Comment:
@rahulraghu94 This log message will be displayed irrespective of whether the current thread is main or child.

Reply:
Isn't that expected behaviour?

Reply:
Oh, sorry. I'm on some meds 📦 
Fixing...